ophyd/v2/core.py

```diff
@@ -788,57 +787,6 @@ async def _wait_for_value():
     await asyncio.wait_for(_wait_for_value(), timeout)
 
 
-@runtime_checkable
-class AsyncReadable(Protocol):
-    """Readable narrowed to only be async"""
-
-    @abstractmethod
-    async def read(self) -> Dict[str, Reading]:
-        ...
-
-    @abstractmethod
-    async def describe(self) -> Dict[str, Descriptor]:
-        ...
-
-
-@dataclass
-class _ReadableRenamer(AsyncReadable, Stageable):
-    readable: AsyncReadable
-    device: Device
-
-    def _rename(self, d: Dict[str, T]) -> Dict[str, T]:
-        return {self.device.name: v for v in d.values()}
-
-    async def read(self) -> Dict[str, Reading]:
-        return self._rename(await self.readable.read())
-
-    async def describe(self) -> Dict[str, Descriptor]:
-        return self._rename(await self.readable.describe())
-
-    def stage(self) -> List[Any]:
-        if isinstance(self.readable, Stageable):
-            return self.readable.stage()
-        else:
-            return []
-
-    def unstage(self) -> List[Any]:
-        if isinstance(self.readable, Stageable):
-            return self.readable.unstage()
-        else:
-            return []
-
-
-@dataclass
-class _UncachedSignal(AsyncReadable):
-    signal: SignalR
-
-    async def read(self) -> Dict[str, Reading]:
-        return await self.signal.read(cached=False)
-
-    async def describe(self) -> Dict[str, Descriptor]:
-        return await self.signal.describe()
```

Comment:
presuming this has been deleted because all signals are cached?

Reply:
Deleted as I found a better way to do this in `StandardReadable` that didn't need this intermediate class

Reply:
ah; this makes a lot of sense. StandardReadable now has cached and uncached signals as part of its definition, making this bit unnecessary. understood.